graphistry/umap_utils.py

```diff
@@ -222,14 +228,33 @@ def umap_fit(self, X: pd.DataFrame, y: Union[pd.DataFrame, None] = None):
         y = self._check_target_is_one_dimensional(y)
         logger.info('-' * 90)
         logger.info(f"Starting UMAP-ing data of shape {X.shape}")
+        # mod_ver=eval(self._umap().__module__.split('.')[0]).__version__
+        if (self.engine == 'cuml') and (self.suffix < 22.06):
```

Comment:
you can make a helper like

```
def is_old_cuml():
   import cuml
   v = cuml.__version__
   if ...
 ```

then this becomes `if (self.engine == 'cuml') and is_old_cuml(): ...`

Reply:
then we import cuml even if its not engine -- I think it has to be nested if -- thats why I used return to break from second if -- so cuml>22.06 runs without knn